code/game/machinery/camera/camera.dm

```diff
@@ -59,13 +60,9 @@
 	if(mapload && z == 1 && prob(3) && !start_active)
 		toggle_cam()
 
-/obj/machinery/camera/Move()
-	remove_from_proximity_list(src, 1)
-	return ..()
-
 /obj/machinery/camera/Destroy()
 	toggle_cam(null, 0) //kick anyone viewing out
-	remove_from_proximity_list(src, 1)
+	QDEL_NULL(proximity_monitor)
```

Comment:
this should probably be on /atom/movable/Destroy, don't even give people the opportunity to forget to gc it.